tests/unit-tests/graphpartitioning/test_metadata.py

```diff
@@ -0,0 +1,731 @@
+        assert (
+            expected_etype_id_map == obj.etype_id_map
+        ), f"etype_id_map initialization failure"
+
+        # Check id <-> etype map.
+        exptected_id_etype_map = {
+            idx: etype for idx, etype in enumerate(etypes)
+        }
+        assert (
+            exptected_id_etype_map == obj.id_etype_map
+        ), f"id_etype_map test failure"
+
+        # Check etypes.
+        expected_etypes = etypes
+        assert expected_etypes == obj.etypes, f"etypes failure."
+
+        # Check global edge ids offsets for etypes.
+        obj._init_global_eid_offsets()
+        counts = np.cumsum([0] + [10 for etype in etypes])
+        ranges = [
+            (counts[idx], counts[idx + 1]) for idx in range(len(counts) - 1)
+        ]
+        expected_geid_offsets = {
+            etype: ranges[idx] for idx, etype in enumerate(etypes)
+        }
+        assert expected_geid_offsets == obj.global_eid_offsets
+
+
+@pytest.mark.parametrize("etypes", [[], [EDGE1], [EDGE1, EDGE2]])
+@pytest.mark.parametrize("edge_fmt", ["csv", "numpy", "parquet"])
+@pytest.mark.parametrize("edge_fmt_del", [" ", "/"])
+@pytest.mark.parametrize("create_files", [True, False])
+@pytest.mark.parametrize("prefix_dir", ["", "edges_"])
+@pytest.mark.parametrize("is_absolute", [True, False])
+def test_etype_files(
+    etypes, edge_fmt, edge_fmt_del, create_files, prefix_dir, is_absolute
+):
+    """Helper function to test various scenarions with edge files.
+    Files presence/absence is test rigorously.
+    """
+    input_dict = {}
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        # Add nodes to the metadata.
+        add_nodes(input_dict, {ntype: 10 for ntype in ntypes})
+
+        file_metadata = namedtuple("file_metadata", "")
+        file_metadata.root_dir = root_dir
+        file_metadata.prefix_dir = prefix_dir
+        file_metadata.create_files = create_files
+        file_metadata.is_absolute = is_absolute
+        file_metadata.file_prefix = "edges_"
+
+        # Add edges and create edge files, if necessary
+        add_edges_files(
+            input_dict,
+            {etype: 10 for etype in etypes},
+            {etype: idx + 1 for idx, etype in enumerate(etypes)},
+            {"name": edge_fmt, "delimiter": edge_fmt_del},
+            file_metadata,
+        )
+
+        # Create metadata.json file.
+        filename = os.path.join(root_dir, METADATA_NAME)
+        with open(filename, "w") as input_handle:
+            json.dump(input_dict, input_handle, indent=4)
+
+        # Test case here.
+        obj = MetadataSchema(METADATA_NAME, root_dir, init_maps=False)
+        obj._init_etype_maps()
+
+        # Check the initialization
+        try:
+            obj._init_etype_files()
+
+            if create_files:
+                for idx, etype in enumerate(etypes):
+                    (file_type, delimiter, data_files) = obj.get_etype_info(
+                        etype
+                    )
```

Comment:
do you really need to break the line for the argument?

Reply:
Done by lintrunner.